port/nordic/nrf5x/src/hal/gpio.zig

```diff
@@ -0,0 +1,85 @@
+const microzig = @import("microzig");
+const peripherals = microzig.chip.peripherals;
+const P0 = peripherals.P0;
+const GPIOTE = microzig.GPIOTE;
+
+pub const Direction = enum(u1) {
+    input = 0x0,
+    output = 0x1,
+};
+
+pub const Sense = enum(u2) {
+    disabled = 0x0,
+    /// sense high level
+    high = 0x2,
+    /// sense low level
+    low = 0x3,
+    _,
+};
+
+pub const Drive = @FieldType(P0.PIN_CNF[0].underlaying_type, "DRIVE");
+
+pub const InputBuffer = enum(u1) {
+    connect = 0x0,
+    disconnect = 0x1,
+};
+
+pub const Pull = enum(u2) {
+    disabled = 0x0,
+    pulldown = 0x1,
+    pullup = 0x3,
+    _,
+};
+
+pub fn num(n: u5) Pin {
+    return @enumFromInt(n);
+}
+
+pub const Pin = enum(u5) {
+    _,
+
+    /// Set GPIO pin as input or output
+    pub inline fn set_direction(pin: Pin, direction: Direction) void {
+        switch (direction) {
+            .input => P0.DIRCLR.raw = pin.mask(),
+            .output => P0.DIRSET.raw = pin.mask(),
+        }
+    }
+    pub inline fn set_sense(pin: Pin, sense: Sense) void {
+        P0.PIN_CNF[@intFromEnum(pin)].modify(.{
```

Comment:
it'll make my PR on top cleaner if you use `get_regs` instead of hardcoding `P0` everywhere.